packages/@aws-cdk/core/lib/stack-synthesizers/bootstrapless-synthesizer.ts

```diff
@@ -39,11 +46,11 @@ export class BootstraplessSynthesizer extends DefaultStackSynthesizer {
     });
   }
 
-  public addFileAsset(_asset: FileAssetSource): FileAssetLocation {
+  public addFileAsset(_asset: FileAssetSource | ExternalFileAssetSource): FileAssetLocation {
```

Comment:
Can't do the type union here either. That's a bit of a bummer though because the only solution would be to either:

* Add a method
* Turn the `FileAssetSource` type into a tagged union, effectively.

We might have to add `addExternalFileAsset()` to a subclass/subinterface of `IStackSynthesizer` and type test for it in the specific asset we're trying to call this from.